assets/src/util/assignment.js

```diff
@@ -8,39 +8,60 @@ const clearGoals = assignments => assignments
     return a
   })
 
-const setAssigmentGoalInputState = (key, assignments, inputFocus) => {
-  return [
-    ...assignments.slice(0, key),
-    {
-      ...assignments[key],
-      inputFocus,
-      inputBlur: !inputFocus
-    },
-    ...assignments.slice(key + 1)
-  ]
+const setAssigmentGoalInputState = (assignmentId, assignments, inputFocus) => {
+  const assignment = assignments.filter(a => a.id === assignmentId)
+  if (assignment.length !== 1) {
+    console.log('Error finding unique assignment id')
+    return assignments
+  } else {
+    const key = assignments.indexOf(assignment[0])
+    return [
+      ...assignments.slice(0, key),
+      {
+        ...assignments[key],
+        inputFocus,
+        inputBlur: !inputFocus
+      },
+      ...assignments.slice(key + 1)
+    ]
+  }
 }
 
-const setAssignmentGoalGrade = (key, assignments, goalGrade) => {
-  return [
-    ...assignments.slice(0, key),
-    {
-      ...assignments[key],
-      goalGrade: goalGrade === '' ? '' : roundToXDecimals(Number(goalGrade), 3),
-      goalGradeSetByUser: !!goalGrade
-    },
-    ...assignments.slice(key + 1)
-  ]
+const setAssignmentGoalGrade = (assignmentId, assignments, goalGrade) => {
+  const assignment = assignments.filter(a => a.id === assignmentId)
+  if (assignment.length !== 1) {
+    console.log('Error finding unique assignment id')
+    return assignments
```

Comment:
Have you encountered this case happening, where the assignment is not found with the `assignmentId` supplied?

Reply:
No, frankly should be impossible

Reply:
The assignmentId being passed in came from a list of assignmentIds for the assignments that the client was already displaying, so there is no reason this should ever happen.  But you know what they say  😛

Reply:
Ah okay, thanks for confirming!